src/read.c

```diff
@@ -4710,18 +4734,20 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         result = avifMetaFindOrCreateItem(meta, newItemID, alphaItem); // Create new empty item.
     }
     if (result != AVIF_RESULT_OK) {
-        avifFree(alphaItemIndices);
+        avifFree(dimgIdxToAlphaItemIdx);
         *isAlphaItemInInput = AVIF_FALSE;
         return result;
     }
     memcpy((*alphaItem)->type, "grid", 4); // Make it a grid and register alpha items as its tiles.
     (*alphaItem)->width = colorItem->width;
     (*alphaItem)->height = colorItem->height;
-    for (uint32_t i = 0; i < alphaItemCount; ++i) {
-        avifDecoderItem * item = meta->items.item[alphaItemIndices[i]];
-        item->dimgForID = (*alphaItem)->id;
+    for (uint32_t dimgIdx = 0; dimgIdx < tileCount; ++dimgIdx) {
+        AVIF_ASSERT_OR_RETURN(dimgIdxToAlphaItemIdx[dimgIdx] != itemIndexNotSet);
```

Comment:
IMPORTANT: We cannot use the `AVIF_ASSERT_OR_RETURN` macro here, because it doesn't free `dimgIdxToAlphaItemIdx` and set the output parameters to `NULL` and `AVIF_FALSE`. We need to use an if statement. Alternatively, use `assert()`.

Reply:
`dimgIdxToAlphaItemIdx[dimgIdx] != itemIndexNotSet` is guaranteed by `alphaItemCount != tileCount` above. I agree an `assert()` would be enough here, but the point of `AVIF_ASSERT_OR_RETURN()` was to replace as many `assert()` as possible, to be safe. I believe a leak is preferable to an out-of-bounds array access.